### PR TITLE
Feature/api/certificate

### PR DESCRIPTION
**Pull request by stargwq**

### CODE REVIEW

File: platform-api/src/main/java/com/flow/platform/api/dao/CredentialDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return "name";
+    }
+
+    @Override
+    public List<Credential> list() {
+        return execute((Session session) -> {
+            CriteriaBuilder builder = session.getCriteriaBuilder();
+            CriteriaQuery<Credential> select = builder.createQuery(Credential.class);
+            Root<Credential> credential = select.from(Credential.class);
+            Predicate condition = builder.not(credential.get("name").isNull());
```
Comment (stargwq) [inline platform-api/src/main/java/com/flow/platform/api/dao/CredentialDaoImpl.java]: list

File: platform-api/src/main/java/com/flow/platform/api/service/CredentialService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.Map;
+
+/**
+ * @author lhl
+ */
+public interface CredentialService {
+
+    Map<String, String> getKeyMap();
+
+    /**
```
Comment (stargwq) [inline platform-api/src/main/java/com/flow/platform/api/service/CredentialService.java]: list

File: platform-api/src/main/java/com/flow/platform/api/controller/CredentialController.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public boolean delete(@PathVariable String name) {
+        return credentialService.delete(name);
+    }
+
+    @PostMapping(path = "/update_credential")
+    public Credential reportStatus(@RequestBody Credential credential) {
+        return credentialService.update(credential);
+    }
+
+    @GetMapping(path = "/get_keys")
```
Comment (yang-guo-2016) [inline platform-api/src/main/java/com/flow/platform/api/controller/CredentialController.java]: change route path

File: platform-api/src/main/java/com/flow/platform/api/domain/Credential.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+import com.flow.platform.api.domain.adaptor.ZonedDateTimeAdaptor;
+import com.flow.platform.domain.Jsonable;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.JsonAdapter;
+import java.time.ZonedDateTime;
+
+/**
+ * @author lhl
+ */
+public class Credential extends Jsonable {
```
Comment (yang-guo-2016) [inline platform-api/src/main/java/com/flow/platform/api/domain/Credential.java]: change to inherit class

File: platform-api/src/main/java/com/flow/platform/api/controller/CredentialController.java (hunk 3 above)
Comment (stargwq) [inline platform-api/src/main/java/com/flow/platform/api/controller/CredentialController.java]: ssh/keys

**Pull request merged**